variables/builtin.go

```diff
@@ -89,6 +90,14 @@ func BuiltinArgs(target domain.Target, platr *platutil.Resolver, gitMeta *gituti
 		// Ensure SOURCE_DATE_EPOCH is always available
 		ret.Add(arg.EarthlySourceDateEpoch, "0")
 	}
+
+	if ftrs.EarthlyCIRunnerArg {
+		value := strings.TrimSpace(os.Getenv(arg.EarthlyCIRunner))
```

Comment:
We typically avoid looking up env vars so deep in the code - we try to only do that in the main package.

You may need to propagate this all the way from main. Here's the rough path:

* Add a hidden flag here: https://github.com/earthly/earthly/blob/main/cmd/earthly/flags.go
* Add a field to this struct and set it to the flag value: https://github.com/earthly/earthly/blob/0857e35cefb1a0155638f47b3ac0558844b48c3f/cmd/earthly/build_cmd.go#L484
* Add a field to this struct and set it to the builder opt field value: https://github.com/earthly/earthly/blob/0857e35cefb1a0155638f47b3ac0558844b48c3f/builder/builder.go#L179
* Add a field to this struct and set it to the converter opt field value: https://github.com/earthly/earthly/blob/0857e35cefb1a0155638f47b3ac0558844b48c3f/earthfile2llb/converter.go#L127

CC @alexcb for a second opinion here, just in case I'm overthinking this.

Reply:
I agree it's a bit more plumbing, but I too would rather have them defined via the flags.

Reply:
sounds good

Reply:
@vladaionescu thanks for guidance!